cloudinit/net/dhcp.py

```diff
@@ -679,6 +703,38 @@ def dhcp_discovery(
             )
             raise NoDHCPLeaseError from error
 
+    @staticmethod
+    def parse_unknown_options_from_packet(
+        data: bytes, number: int
```

Comment:
nit: lets make this param name a bit more specific?
```suggestion
        data: bytes, dhcp_option_number: int
```